src/modules/cmdpal/Microsoft.CmdPal.UI/ShellPage.xaml.cs

```diff
@@ -37,6 +40,9 @@ public ShellPage()
         WeakReferenceMessenger.Default.Register<NavigateToDetailsMessage>(this);
         WeakReferenceMessenger.Default.Register<PerformCommandMessage>(this);
 
+        WeakReferenceMessenger.Default.Register<ShowDetailsMessage>(this);
+        WeakReferenceMessenger.Default.Register<HideDetailsMessage>(this);
```

Comment:
To do this the more XAML-y way and avoid all the mucking of controls in the code-behind below (generally a thing we try and avoid if we can)...

We could expose a `IsDetailsVisible` property in our VM and then have the logic which controls that based on receiving these messages there.

Then in XAML we can define a VisualStateManager which triggers off the boolean value of that property (we have IsEqualsStateTrigger in the Toolkit, but for bool you can just bind to `IsActive`, iirc)

Then the VSM can muck with the grid columns and visibility, etc... Should make it easier to deal with transitions and other stuff as well in the future for animations and other things, though @niels9001 knows that a bit better. 😅

Reply:
Okay - I think I followed, and yet, XAML refuses to let me debug it ☹️ 

3ad8b6bbf is a commit that by all accounts, should do everything right, and I have no idea why it doesn't

Reply:
As discussed, we had to move the VSM into the root-level Grid vs. being at the top-level Page level.

For future reference: https://learn.microsoft.com/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.visualstatemanager#examples